app/lib/theme/CustomFields.tsx

```diff
@@ -0,0 +1,212 @@
+  <>
+    <span id={id && `${id}-diffOld`} className="diffOld">
+      <NumberFormat
+        thousandSeparator
+        prefix={isMoney ? "$" : ""}
+        id={id}
+        displayType="text"
+        value={oldData}
+      />
+    </span>
+    &nbsp;
+    <span>
+      <strong>
+        <em>REMOVED</em>
+      </strong>
+    </span>
+  </>
+);
+
+const CUSTOM_FIELDS: Record<string, React.FunctionComponent<FieldProps>> = {
+  StringField: (props) => {
+    const { idSchema, formData, formContext } = props;
+    const id = idSchema?.$id;
+    const previousValue = formContext?.oldData?.[props.name];
+
+    if (previousValue && formData) {
+      return showStringDiff(id, previousValue, formData);
+    } else if (
+      !previousValue &&
+      formData &&
+      formContext.operation === "CREATE"
+    ) {
+      return showStringAdded(id, formData);
+    } else if (
+      formContext.operation === "ARCHIVE" ||
+      (!formData && previousValue)
+    ) {
+      return showStringRemoved(id, previousValue);
+    } else {
+      return <>DISPLAY ERROR</>;
+    }
+  },
+  NumberField: (props) => {
+    const { idSchema, formData, formContext, uiSchema } = props;
+    const id = idSchema?.$id;
+    const previousValue = formContext?.oldData?.[props.name];
+
+    if (uiSchema["ui:options"]) {
+      if (previousValue && formData && formContext.operation === "UPDATE") {
+        return showStringDiff(
+          id,
+          formContext?.oldUiSchema?.[props.name]?.["ui:options"]?.text,
+          uiSchema["ui:options"].text as string
+        );
+      } else if (
+        !previousValue &&
+        formData &&
+        formContext.operation === "CREATE"
+      ) {
+        return showStringAdded(id, uiSchema["ui:options"].text as string);
+      } else if (
+        formContext.operation === "ARCHIVE" ||
+        (!formData && previousValue)
+      ) {
+        return showStringRemoved(
+          id,
+          formContext?.oldUiSchema?.[props.name]?.["ui:options"]?.text
+        );
+      } else {
+        return <>DISPLAY ERROR</>;
+      }
+    } else {
+      if (previousValue && formData && formContext.operation === "UPDATE") {
+        return showNumberDiff(
+          id,
+          formContext?.oldData?.[props.name],
+          formData,
+          uiSchema?.["ui:widget"] === "MoneyWidget"
+        );
+      } else if (
+        !previousValue &&
+        formData &&
+        formContext.operation === "CREATE"
+      ) {
+        return showNumberAdded(
+          id,
+          formData,
+          uiSchema?.["ui:widget"] === "MoneyWidget"
+        );
+      } else if (
+        formContext.operation === "ARCHIVE" ||
+        (!formData && previousValue)
+      ) {
+        return showNumberRemoved(
+          id,
+          formContext?.oldData?.[props.name],
+          uiSchema?.["ui:widget"] === "MoneyWidget"
+        );
+      } else {
+        return <>DISPLAY ERROR</>;
+      }
+    }
+  },
+};
```

Comment:
This is some pretty intricate display logic, and could definitely use a few unit tests